### PR TITLE
feat(queue): make some parts free

### PR DESCRIPTION
**Pull request by sileht**
This makes queue action free, but disable some options by default:
merge_bot_account, priority, speculative_checks.

This changes the behavior of priority feature, even for merge action.
Now it's reported as failure, instead of replacing the configuration by
medium.

Fixes MRGFY-359

### CODE REVIEW

File: mergify_engine/actions/queue.py [hunk 1]
```diff
@@ -58,18 +58,51 @@ class QueueAction(merge_base.MergeBaseAction):
...
                     ctxt.pull["base"]["repo"]["owner"]["login"]
                 ),
             )
 
+        if self.queue_rule.config[
+            "speculative_checks"
+        ] > 1 and not ctxt.subscription.has_feature(subscription.Features.QUEUE_ACTION):
+            return check_api.Result(
+                check_api.Conclusion.ACTION_REQUIRED,
+                "Queue with `speculative_checks` set are disabled.\n"
```
Comment (jd) [inline mergify_engine/actions/queue.py:91]:
```suggestion
                "Queues with `speculative_checks` set are disabled.\n"
```

File: mergify_engine/actions/queue.py [hunk 2]
```diff
@@ -58,18 +58,51 @@ class QueueAction(merge_base.MergeBaseAction):
...
+                ),
+            )
+
+        if self.config["merge_bot_account"] and not ctxt.subscription.has_feature(
+            subscription.Features.MERGE_BOT_ACCOUNT
+        ):
             return check_api.Result(
                 check_api.Conclusion.ACTION_REQUIRED,
-                "Queue action is disabled",
+                "Queue with `merge_bot_account` set are disabled",
```
Comment (jd) [inline mergify_engine/actions/queue.py:80]:
```suggestion
                "Queues with `merge_bot_account` set are disabled",
```

File: mergify_engine/actions/queue.py [hunk 3]
```diff
@@ -58,18 +58,51 @@ class QueueAction(merge_base.MergeBaseAction):
...
-        if not ctxt.subscription.has_feature(subscription.Features.QUEUE_ACTION):
+    def _subscription_status(
+        self, ctxt: context.Context
+    ) -> typing.Optional[check_api.Result]:
+        if self.config["update_bot_account"] and not ctxt.subscription.has_feature(
+            subscription.Features.MERGE_BOT_ACCOUNT
+        ):
+            return check_api.Result(
+                check_api.Conclusion.ACTION_REQUIRED,
+                "Queue with `update_bot_account` set are disabled",
```
Comment (jd) [inline mergify_engine/actions/queue.py:69]:
```suggestion
                "Queueswith `update_bot_account` set are disabled",
```

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/actions/queue.py (hunk 1 above)
Comment (jd) [reply mergify_engine/actions/queue.py:91]: Also: s/disabled/unavailable/ ?

File: mergify_engine/actions/merge.py [hunk 4]
```diff
@@ -70,6 +72,55 @@ def validate_config(self, mergify_config: "rules.MergifyConfig") -> None:
...
+    def _subscription_status(
+        self, ctxt: context.Context
+    ) -> typing.Optional[check_api.Result]:
+
+        if self.config["update_bot_account"] and not ctxt.subscription.has_feature(
+            subscription.Features.MERGE_BOT_ACCOUNT
+        ):
+            return check_api.Result(
+                check_api.Conclusion.ACTION_REQUIRED,
+                "Merge with `update_bot_account` set are unavailable",
```
Comment (jd) [inline mergify_engine/actions/merge.py:84]: ditto, if you use `are` we need `Merges`
or just go for `is` :)

**Review by jd — CHANGES_REQUESTED** [review]

File: mergify_engine/actions/merge.py (hunk 4 above)
Comment (sileht) [reply mergify_engine/actions/merge.py]: Go for `is` then

**Review by jd — APPROVED** [review]

**Pull request merged**